Documentation/kubernetes/install/kubespray.rst

```diff
@@ -10,7 +10,171 @@
+test purposes. By default, this tutorial will create:
+
+  - VPC with 2 public and private subnets
+  - Bastion Hosts and NAT Gateways in the Public Subnet
+  - Three of each (masters, etcd, and worker nodes) in the Private Subnet
+  - AWS ELB in the Public Subnet for accessing the Kubernetes API from
+    the internet
+  - Terraform scripts use ``CoreOS`` as base image.
+
+Example ``terraform.tfvars`` file:
+
+.. code:: bash
+
+  #Global Vars
+  aws_cluster_name = "kubespray"
+
+  #VPC Vars
+  aws_vpc_cidr_block = "XXX.XXX.192.0/18"
+  aws_cidr_subnets_private = ["XXX.XXX.192.0/20","XXX.XXX.208.0/20"]
+  aws_cidr_subnets_public = ["XXX.XXX.224.0/20","XXX.XXX.240.0/20"]
+
+  #Bastion Host
+  aws_bastion_size = "t2.medium"
+
+
+  #Kubernetes Cluster
+
+  aws_kube_master_num = 3
+  aws_kube_master_size = "t2.medium"
+
+  aws_etcd_num = 3
+  aws_etcd_size = "t2.medium"
+
+  aws_kube_worker_num = 3
+  aws_kube_worker_size = "t2.medium"
+
+  #Settings AWS ELB
+
+  aws_elb_api_port = 6443
+  k8s_secure_api_port = 6443
+  kube_insecure_apiserver_address = "0.0.0.0"
+
+
+-----------------------
+Apply the configuration
+-----------------------
+
+``terraform init`` to initialize the following modules
+
+  - ``module.aws-vpc``
+  - ``module.aws-elb``
+  - ``module.aws-iam``
+
+.. code:: bash
+
+  $ cd contrib/terraform/aws
+  $ terraform init
+
+Once Initialized , execute:
+
+.. code:: bash
+
+  $ terraform plan -out=aws_kubespray_plan
+
+This will generate a file ``aws_kubespray_plan`` depicting an execution
+plan of the infrastructure that will be created on AWS. To apply, execute:
+
+.. code:: bash
+
+  $ terraform init
+  $ terraform apply "aws_kubespray_plan"
+
+Terraform automatically creates an Ansible Inventory file at ``inventory/hosts``.
+
+Installing Kubernetes cluster with Cilium as CNI
+================================================
+
+Kubespray uses Ansible as its substrate for provisioning and orchestration. Once the infrastructure is created, you can run the ansible playbook to install Kubernetes and all the required dependencies. Execute the below command in the kubespray clone repo, providing the correct path of the AWS EC2 ssh private key in ``ansible_ssh_private_key_file=~/path/to/ec2-key-file.pem``
+
+If you want to change the current default version of Cilium, then edit ``roles/download/defaults/main.yml``.
+
+.. code:: bash
+
+  $ ansible-playbook -i ./inventory/hosts ./cluster.yml -e ansible_user=core -e bootstrap_os=coreos -e kube_network_plugin=cilium -b --become-user=root --flush-cache  -e ansible_ssh_private_key_file=~/path/to/ec2-key-file.pem
+
+
+Validate the cluster
+====================
+
+To check if cluster is created successfully, ssh into the bastion host with the user ``core``. 
+
+.. code:: bash
+
+  # Get information about the basiton host 
+  $ cat ssh-bastion.conf    
+  $ ssh -i ~/path/to/ec2-key-file.pem core@public_ip_of_bastion_host 
```

Comment:
Not too set on this, but I usually put things which users have to substitute within angle brackets with a description, e.g. `<path to EC2 SSH Key file>`